### ISSUE TITLE
dap-julia: Adapter for Julialang

### ISSUE THREAD

**Issue by fleimgruber** [emacs-lsp/dap-mode#173]
The implementation I found is https://github.com/julia-vscode/DebugAdapter.jl. Could this be wrapped with a `dap-julia.el`? AFAICS, the dap-mode adapters use external programs so one would have to be written in Julia, correct?

**Comment by yyoncho** [comment]
Yes. You need the debug adapter and the corresponding debug templates.

**Comment by fleimgruber** [comment]
After more reading, it seems that the repo I linked to is just the interface code and not a proper debug server. That server still needs to be written, see also https://github.com/julia-vscode/julia-vscode/issues/125

**Comment by jbrnd** [comment]
Some further comments here: https://github.com/puremourning/vimspector/issues/256#issuecomment-695794837

**Comment by zot** [comment]
Looks like Julia's DAP server works with VSCode now...

**Comment by zot** [comment]
Looks like Julia's DAP server works with VSCode now...

**Comment by zot** [comment]
Looks like Julia's DAP server works with VSCode now...

**Comment by zot** [comment]
Looks like Julia's DAP server works with VSCode now...

**Comment by fleimgruber** [comment]
@zot thanks for the update. What is missing here is a standalone DAP server executable, so did you manage to use it outsider of VS Code?

**Comment by zot** [comment]
Not yet -- just found out about it and saw this issue so I thought I'd post a little info here. I don't know how standalone this is:

https://github.com/julia-vscode/julia-vscode

**Comment by fleimgruber** [comment]
This is what I did in https://github.com/emacs-lsp/dap-mode/issues/173#issuecomment-567208038. In the meantime the Julia implementation was done (your link to `run_debugger.jl`), but the comment linked in https://github.com/emacs-lsp/dap-mode/issues/173#issuecomment-817454492 is still valid I fear.

**Comment by Nidish96** [comment]
Has there been any progress on this yet?
